service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/status/service/EhrStatusService.java

```diff
@@ -0,0 +1,188 @@
+            //            ackPendingOptional
+            //                .map(ackPending -> ackPending.getTypeCode().equals("AA"))
+            //                .orElse(false)
+            //
+            //                && ackToRequestorOptional
+            //                .map(ackToRequester -> ackToRequester.getTypeCode().equals("AA"))
+            //                .orElse(false)
+            //
+            errorOptional.isEmpty()
+
+                && receivedAcknowledgementOptional
+                .map(acknowledgement ->
+                    Optional.ofNullable(acknowledgement.getErrors()).isPresent())
+                .orElse(false)) {
+
+            return FAILED_INCUMBENT;
+        }
+
+        return IN_PROGRESS;
+    }
+
+    private List<EhrStatus.AttachmentStatus> getAttachmentStatusList(EhrExtractStatus record,
+        List<EhrExtractStatus.EhrReceivedAcknowledgement> acknowledgements) {
+
+        List<EhrStatus.AttachmentStatus> attachmentStatusList = new ArrayList<>();
+
+        var accessDocumentOptional = Optional.ofNullable(record.getGpcAccessDocument());
+
+        accessDocumentOptional.ifPresent(accessDocument ->
+            accessDocument.getDocuments().forEach(gpcDocument ->
+                attachmentStatusList.add(
+                    EhrStatus.AttachmentStatus.builder()
+                        .identifier(gpcDocument.getIdentifier())
+                        .fileName(gpcDocument.getFileName())
+                        .fileStatus(getFileStatus(gpcDocument, acknowledgements))
+                        .originalDescription(gpcDocument.getOriginalDescription())
+                        .build())
+            )
+        );
+
+        return attachmentStatusList.stream()
+            .filter(attachmentStatus -> attachmentStatus.getFileStatus() != SKELETON_MESSAGE)
+            .collect(Collectors.toList());
+    }
+
+    public FileStatus getFileStatus(EhrExtractStatus.GpcDocument document,
+        List<EhrExtractStatus.EhrReceivedAcknowledgement> acknowledgements) {
+
+        Optional<String> fileNameOptional = Optional.ofNullable(document.getFileName());
+        Optional<String> objectNameOptional = Optional.ofNullable(document.getObjectName());
+        Optional<EhrExtractStatus.GpcAccessDocument.SentToMhs> sentToMhsOptional = Optional.ofNullable(document.getSentToMhs());
+
+        if (document.isSkeleton()) {
+            return SKELETON_MESSAGE;
+        }
+
+        if (sentToMhsOptional.isPresent()) {
+            List<String> messageIds = sentToMhsOptional.get().getMessageId();
+            boolean documentHasNack = acknowledgements.stream()
+                .filter(ack -> ack.getErrors() != null)
+                .anyMatch(ack -> messageIds.contains(ack.getMessageRef()));
+
+            if (documentHasNack) {
+                return ERROR;
+            }
+        }
+
+        if (objectNameOptional.isPresent() && objectNameOptional.get().contains("AbsentAttachment")) {
```

Comment:
Minor improvement, we have the same logic for two different variables, extract the variable first and then run the if logic. Story coding example..

boolean isPlaceholderFile = (objectNameOptional.isPresent() && objectNameOptional.get().contains("AbsentAttachment")) || (fileNameOptional.isPresent() && fileNameOptional.get().contains("AbsentAttachment"))

if (isPlaceholderFile) { reurtrPlaceholder}

Reply:
Could even extract the two checks into their own function to tidy up further

    boolean containsPlaceholder = isPlaceholderFile(object) || isPlaceholderFile(file)
    if (containsPlaceholder) { return PLACEHOLDER } 

    isPlaceholderFile(fileObject) {
    
    if (fileObject is None) {
        Return False
    }
    
    if (!fileObject.isPresent ect..){
        Return True    
    }
    return True
    }